server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -3747,28 +3747,34 @@ ReplicationTracker getReplicationTracker() {
      * Executes a scheduled refresh if necessary. Completes the listener with true if a refreshed was performed otherwise false.
      */
     public void scheduledRefresh(ActionListener<Boolean> listener) {
-        verifyNotClosed();
-        boolean listenerNeedsRefresh = refreshListeners.refreshNeeded();
-        final Engine engine = getEngine();
-        if (isReadAllowed() && (listenerNeedsRefresh || engine.refreshNeeded())) {
-            if (listenerNeedsRefresh == false // if we have a listener that is waiting for a refresh we need to force it
-                && engine.allowSearchIdleOptimization()
-                && isSearchIdle()
-                && indexSettings.isExplicitRefresh() == false
-                && active.get()) { // it must be active otherwise we might not free up segment memory once the shard became inactive
-                // lets skip this refresh since we are search idle and
-                // don't necessarily need to refresh. the next searcher access will register a refreshListener and that will
-                // cause the next schedule to refresh.
-                engine.maybePruneDeletes(); // try to prune the deletes in the engine if we accumulated some
-                setRefreshPending(engine);
-                ActionListener.completeWith(listener, () -> false);
-            } else {
-                logger.trace("refresh with source [schedule]");
-                engine.maybeRefresh("schedule", listener.map(Engine.RefreshResult::refreshed));
+        try {
```

Comment:
nit You could replace the try/catch with ActionListener.run().

Reply:
I pushed 50940c26fd7d6fd9190a38994aec7700a0ab99b9